python/IAM_POLICY_REQUIRED/IAM_POLICY_REQUIRED.py

```diff
@@ -0,0 +1,526 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+ Rule Name:
+   IAM_POLICY_REQUIRED
+ Description:
+   To check IAM users and roles have a given policy attached directly or through a group.
+ Trigger:
+   Configuration Change on AWS::IAM::User/AWS::IAM::Role
+ Reports on:
+   AWS::IAM::User,AWS::IAM::Role
+ Rule Parameters:
+   | ---------------------- | --------- | -------------------------------------------------------- |
+   | Parameter Name         | Type      | Description                                              |
+   | ---------------------- | --------- | -------------------------------------------------------- |
+   | policyArns             | Required  | ARNs of the policies which should be attached to all     |
+   |                        |           | users and roles.                                         |
+   | ---------------------- | --------- | -------------------------------------------------------- |
+   | exceptionList          | Optional  | Represents the IAM users and roles which are exempted    |
+   |                        |           | from the IAM Config rule. The valid users in this list   |
+   |                        |           | will be compliant by default.                            |
+   |                        |           | List of all the User Names separated by a comma          |
+   | ---------------------- | --------- | -------------------------------------------------------- |
+ Feature:
+   As: a Security Officer
+   I want: To ensure that IAM roles and users have mandatory policies attached
+   In order to: enforce mandatory permissions for IAM entities
+ Scenarios:
+  Scenario: 1
+     Given: No IAM Users or Roles exist
+      When: Evaluation occurs

```

Comment:
I'd remove the "when" statement in all scenarios. While it is true, I'd never saw it requested by anyone to understand the gherkin.

Reply:
Agree, just the WHEN statement is always the same in this case, hence the fact that it has no value here.